src/core_plugins/kibana/public/management/sections/settings/components/component_registry.js

```diff
@@ -0,0 +1,35 @@
+/*
```

Comment:
it is my understanding that we can't import things from OSS from x-pack by using relative paths, e.g. `import { registerComponent } from '../../../src/core_plugins/kibana/[...]'` won't work outside of dev mode.

you can probably get around this by passing it through via `src/ui/public/management/index.js`:
```js
export { registerComponent } from '../../../core_plugins/kibana/[...]'`;
```
then it can be used from within x-pack:
```js
import { registerComponent } from 'ui/management';
```

given this method, the function names would be better as `registerSettingsComponent` and `getSettingsComponent` to be explicit about the context once the usage is outside of the `settings/` directory

Reply:
Excellent point, this is something I clearly hadn't considered. Thank you!